ui/src/api.ts

```diff
@@ -123,3 +117,36 @@ async function doGet<T>(get: Promise<AxiosResponse<T>>) {
   }
   return null;
 }
+
+/** 
+ * helper function for slicing arrays.
+ */
+function sliceArray(array: string[], size: number) {
```

Comment:
@jocelynzsun can we name this something like "splitArray" instead? Slice is an operation that already has a well-defined, different set of behavior than this (see `Array.slice` documentation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)).

Because we're not calling it elsewhere, actually, maybe we should just get rid of this function and include the code itself in the `collectPromise` method, to keep from unnecessary code fragmentation?